### PR TITLE
8340354: Open source AWT desktop properties and print related tests

### PR DESCRIPTION
**Pull request by kumarabhi006**
AWT Desktop properties and printer related tests are converted from applet to manual and moved to open.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8340354](https://bugs.openjdk.org/browse/JDK-8340354): Open source AWT desktop properties and print related tests (**Bug** - P4)

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21197/head:pull/21197` \
`$ git checkout pull/21197`

Update a local copy of the PR: \
`$ git checkout pull/21197` \
`$ git pull https://git.openjdk.org/jdk.git pull/21197/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21197`

View PR using the GUI difftool: \
`$ git pr show -t 21197`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21197.diff">https://git.openjdk.org/jdk/pull/21197.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21197#issuecomment-2375997848)

### CODE REVIEW

File: test/jdk/java/awt/PrintJob/PrintComponentTest.java [hunk 1]
```diff
@@ -0,0 +1,484 @@
...
+        MenuItem itemJFC =
+                new MenuItem("Print to Printer with Cross-Platform Dialog...");
+        itemJFC.setActionCommand("common");
+        MenuItem itemNative =
+                new MenuItem("Print to Printer with Native Dialog...");
+        itemNative.setActionCommand("native");
+        MenuItem itemBackground =
+                new MenuItem("Print to Printer in Background");
+        itemBackground.setActionCommand("none");
+        MenuItem itemScreen  = new MenuItem("Print to Screen...");
```
Comment (turbanoff) [inline test/jdk/java/awt/PrintJob/PrintComponentTest.java:227]:
```suggestion
        MenuItem itemScreen = new MenuItem("Print to Screen...");
```

File: test/jdk/java/awt/PrintJob/PrintComponentTest.java [hunk 2]
```diff
@@ -0,0 +1,484 @@
...
+            jobAttributes.setDialog(dialog);
+            pageAttributes.setOrigin(OriginType.PRINTABLE);
+            System.err.println(jobAttributes);
+            System.err.println(pageAttributes);
+
+            PassFailJFrame.log("About to show print dialog...");
+
+            PrintJob pj = getToolkit().getPrintJob(
+                    MainFrame.this, "Print test!", jobAttributes, pageAttributes);
+            if( pj == null ) return;
```
Comment (turbanoff) [inline test/jdk/java/awt/PrintJob/PrintComponentTest.java:270]:
```suggestion
            if (pj == null) return;
```

File: test/jdk/java/awt/PrintJob/PrintComponentTest.java [hunk 3]
```diff
@@ -0,0 +1,484 @@
...
+            //g.dispose();
+            // end 4212564
+
+            pj.end();
+        }
+    }
+
+    class ActionPrintToScreen implements ActionListener {
+        public void actionPerformed(ActionEvent ev) {
+            PrintFrame  printFrame = new PrintFrame( MainFrame.this );
```
Comment (turbanoff) [inline test/jdk/java/awt/PrintJob/PrintComponentTest.java:312]:
```suggestion
            PrintFrame printFrame = new PrintFrame( MainFrame.this );
```

File: test/jdk/java/awt/PrintJob/PrintComponentTest.java [hunk 4]
```diff
@@ -0,0 +1,484 @@
...
+
+        public void paint( Graphics g ) {
+            printComponent.printAll(g);
+        }
+    }
+
+    class LWButton extends Component {
+        String label;
+        int     width = 100;
+        int     height = 30;
```
Comment (turbanoff) [inline test/jdk/java/awt/PrintJob/PrintComponentTest.java:350]:
```suggestion
        int width = 100;
        int height = 30;
```

File: test/jdk/java/awt/PrintJob/PrintComponentTest.java [hunk 5]
```diff
@@ -0,0 +1,484 @@
...
+            g.fillRect(0, 0, d.width, d.height);
+            g.setColor(Color.black);
+            int x = 5;
+            int y = (d.height - 5);
+            g.drawString(label, x, y);
+        }
+
+        public Dimension getPreferredSize()
+        {
+            return new Dimension(width,height);
```
Comment (turbanoff) [inline test/jdk/java/awt/PrintJob/PrintComponentTest.java:370]:
```suggestion
            return new Dimension(width, height);
```

File: test/jdk/java/awt/PrintJob/PrintComponentTest.java [hunk 6]
```diff
@@ -0,0 +1,484 @@
...
+                    3);
+            // swap third and fourth to correct positions
+            remove(third);
+            add(third, 2);
+            // re-validate so third and fourth peers change position
+            validate();
+            // now make things really interesting with a lightweight
+            // component at the top of the z-order, that should print
+            // _below_ the native guys to match the screen...
+            add(new LWButton("LWButton"),0);
```
Comment (turbanoff) [inline test/jdk/java/awt/PrintJob/PrintComponentTest.java:454]:
```suggestion
            add(new LWButton("LWButton"), 0);
```

File: test/jdk/java/awt/PrintJob/PrintCompatibilityTest.java [hunk 7]
```diff
@@ -0,0 +1,446 @@
...
+        Menu menu = new Menu("Print");
+        Menu menu2 = new Menu("File");
+        Menu menu3 = new Menu("Edit");
+        Menu menu4 = new Menu("ReallyReallyReallyReallyReallyReallyReallyReallyReallyReallyReallyReallyReallyReallyReallyReallyReallyReallyReallyReallyReallyReallyReallyReallyReallyReallyReallyReallyReallyReallyLong");
+        menu2.setFont(new Font("SansSerif", Font.BOLD, 20));
+        menu2.setEnabled(false);
+        menu3.setFont(new Font("Monospaced", Font.ITALIC, 18));
+        menu3.setEnabled(false);
+        menu4.setEnabled(false);
+        MenuItem itemPrinter  = new MenuItem("Print to Printer...");
```
Comment (turbanoff) [inline test/jdk/java/awt/PrintJob/PrintCompatibilityTest.java:195]:
```suggestion
        MenuItem itemPrinter = new MenuItem("Print to Printer...");
```

File: test/jdk/java/awt/PrintJob/PrintCompatibilityTest.java [hunk 8]
```diff
@@ -0,0 +1,446 @@
...
+        MenuItem itemScreen  = new MenuItem("Print to Screen...");
```
Comment (turbanoff) [inline test/jdk/java/awt/PrintJob/PrintCompatibilityTest.java:196]:
```suggestion
        MenuItem itemScreen = new MenuItem("Print to Screen...");
```

File: test/jdk/java/awt/PrintJob/PrintCompatibilityTest.java [hunk 9]
```diff
@@ -0,0 +1,446 @@
...
+
+        itemPrinter.addActionListener( new ActionPrint() );
+        itemScreen.addActionListener( new ActionPrintToScreen() );
+        setVisible(true);
+    }
+
+    static void printProps(Properties props)
+    {
+        Enumeration propNames = props.propertyNames();
+        while(propNames.hasMoreElements()) {
```
Comment (turbanoff) [inline test/jdk/java/awt/PrintJob/PrintCompatibilityTest.java:214]:
```suggestion
        while (propNames.hasMoreElements()) {
```

File: test/jdk/java/awt/PrintJob/PrintCompatibilityTest.java [hunk 10]
```diff
@@ -0,0 +1,446 @@
...
+    class ActionPrint implements ActionListener {
+        private final int ITERATIONS = 1;
+        private Properties props = new Properties();
+
+        public void actionPerformed(ActionEvent ev) {
+            PassFailJFrame.log("About to show print dialog...");
+            printProps(props);
+            PrintJob pj = getToolkit().getPrintJob(
+                MainFrame.this, "Print test!", props);
+            if( pj == null ) {
```
Comment (turbanoff) [inline test/jdk/java/awt/PrintJob/PrintCompatibilityTest.java:229]:
```suggestion
            if (pj == null) {
```

File: test/jdk/java/awt/PrintJob/PrintCompatibilityTest.java [hunk 11]
```diff
@@ -0,0 +1,446 @@
...
+            //g.dispose();
+            // end 4212564
+
+            pj.end();
+        }
+    }
+
+    class ActionPrintToScreen implements ActionListener {
+        public void actionPerformed(ActionEvent ev) {
+            PrintFrame  printFrame = new PrintFrame( MainFrame.this );
```
Comment (turbanoff) [inline test/jdk/java/awt/PrintJob/PrintCompatibilityTest.java:274]:
```suggestion
            PrintFrame printFrame = new PrintFrame( MainFrame.this );
```

File: test/jdk/java/awt/PrintJob/PrintCompatibilityTest.java [hunk 12]
```diff
@@ -0,0 +1,446 @@
...
+
+        public void paint( Graphics g ) {
+            printComponent.printAll(g);
+        }
+    }
+
+    class LWButton extends Component {
+        String label;
+        int     width = 100;
+        int     height = 30;
```
Comment (turbanoff) [inline test/jdk/java/awt/PrintJob/PrintCompatibilityTest.java:312]:
```suggestion
        int width = 100;
        int height = 30;
```

File: test/jdk/java/awt/PrintJob/PrintCompatibilityTest.java [hunk 13]
```diff
@@ -0,0 +1,446 @@
...
+            g.fillRect(0, 0, d.width, d.height);
+            g.setColor(Color.black);
+            int x = 5;
+            int y = (d.height - 5);
+            g.drawString(label, x, y);
+        }
+
+        public Dimension getPreferredSize()
+        {
+            return new Dimension(width,height);
```
Comment (turbanoff) [inline test/jdk/java/awt/PrintJob/PrintCompatibilityTest.java:332]:
```suggestion
            return new Dimension(width, height);
```

File: test/jdk/java/awt/PrintJob/PrintCompatibilityTest.java [hunk 14]
```diff
@@ -0,0 +1,446 @@
...
+            g.drawLine(0, d.height - 1, 0, 0);
+            g.setColor(Color.black);
+            int x = 5;
+            int y = (d.height - 5);
+            g.drawString(label, x, y);
+        }
+
+        public Dimension getPreferredSize()
+        {
+            return new Dimension(width,height);
```
Comment (turbanoff) [inline test/jdk/java/awt/PrintJob/PrintCompatibilityTest.java:364]:
```suggestion
            return new Dimension(width, height);
```

File: test/jdk/java/awt/PrintJob/PrintCompatibilityTest.java [hunk 15]
```diff
@@ -0,0 +1,446 @@
...
+                            3);
+            // swap third and fourth to correct positions
+            remove(third);
+            add(third, 2);
+            // re-validate so third and fourth peers change position
+            validate();
+            // now make things really interesting with a lightweight
+            // component at the top of the z-order, that should print
+            // _below_ the native guys to match the screen...
+            add(new LWButton("LWButton"),0);
```
Comment (turbanoff) [inline test/jdk/java/awt/PrintJob/PrintCompatibilityTest.java:416]:
```suggestion
            add(new LWButton("LWButton"), 0);
```

File: test/jdk/java/awt/DesktopProperties/ThreeDBackgroundColor.java [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Color;
+        import java.awt.Frame;
```
Comment (prrace) [inline test/jdk/java/awt/DesktopProperties/ThreeDBackgroundColor.java:25]: seem to have some extra spaces here

**Review by prrace — APPROVED** [review]

File: test/jdk/java/awt/DesktopProperties/ThreeDBackgroundColor.java (hunk 16 above)
Comment (kumarabhi006) [reply test/jdk/java/awt/DesktopProperties/ThreeDBackgroundColor.java]: Updated.

**Review by prrace — APPROVED** [review]

**Pull request closed without merging**